src/ptychocg/solver.py

```diff
@@ -53,8 +53,13 @@ class PtychoCuFFT(ptychofft):
         simultaneously.
     """
 
-    def __init__(self, nscan, nprb, ndetx, ndety, ntheta, nz, n, ptheta):
+    def __init__(self, nscan, nprb, ndetx, ndety, ntheta, nz, n, ptheta, igpu):
         """Please see help(PtychoCuFFT) for more info."""
+        cp.cuda.Device(igpu).use()  # gpu id to use
+        # set cupy to use unified memory
+        pool = cp.cuda.MemoryPool(cp.cuda.malloc_managed)
+        cp.cuda.set_allocator(pool.malloc)
+
```

Comment:
I'm not sure that moving this code into the class is a good idea because it always overrides any memory settings that the end-user may want.